src/mca/gds/hash/gds_hash.c

```diff
@@ -303,6 +311,89 @@ static pmix_job_t* get_tracker(const pmix_nspace_t nspace, bool create)
     return trk;
 }
 
+static bool check_hostname(char *h1, char *h2)
+{
+    size_t len1, len2;
+
+    /* only check to the shortest name */
+    len1 = strlen(h1);
+    len2 = strlen(h2);
+    if (len2 < len1) {
+        len1 = len2;
+    }
+
+    if (0 == strncmp(h1, h2, len1)) {
```

Comment:
I don't think this is correct. We have clusters where the nodes are numbered like:
```
node1
node10
node100
```
With this logic these three different nodes would all match each other.

Reply:
Shame on you!! You folks should know better than to do something like that 😦 - I believe that was taught in sys admin 101. 😄 

I suppose we could look at the name for '.' and only check to that point, though it still leaves open the question of names that start the same but differ by domain. Perhaps the only real solution is to pass down an attribute telling us what to do with domain names - i.e., to include them in tests or strip?